src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -1294,7 +1294,7 @@ def _construct_app_and_env_for_source_or_repo(self):
         image = None if self.get_argument_image().__eq__(HELLO_WORLD_IMAGE) else _reformat_image(self.get_argument_source(), self.get_argument_repo(), self.get_argument_image())
 
         has_dockerfile = _has_dockerfile(self.get_argument_source(), dockerfile)
-        if not self.get_argument_source() or has_dockerfile:
```

Comment:
Would you mind explaining how the changes made in this PR are enabling these commands to build a provided GitHub repository without a Dockerfile?

Reply:
+1, I would appreciate a bit more explanation as well as the changes are not super self-evident.

Reply:
Thank you @daniv-msft  and @cormacpayne for taking a look! 
As of today, az containerapp up/create w --repo throws a validation error if there was no Dockerfile in the repo [here](https://github.com/snehapar9/azure-cli-extensions/blob/main/src/containerapp/azext_containerapp/_up_utils.py#L718)

Removed the condition to check for `not self.get_argument_source()` i.e. `repo` from this line because it is not necessary to check if the customer's repo contains a Dockerfile here. The `container-app-deploy-action` looks for `Dockerfile` by default first before attempting to use the builder to build the app.

But, if folks feel we should check if repo contains `Dockerfile` here, without solely relying on the GitHub action, I can modify `_has_dockerfile` to  determine if either source or repo contains `Dockerfile` and remove [this validation](https://github.com/snehapar9/azure-cli-extensions/blob/main/src/containerapp/azext_containerapp/_up_utils.py#L718) validation instead.

Reply:
OK I'm just now looking up the [`_get_dockerfile_content`](https://github.com/Azure/azure-cli-extensions/blob/main/src/containerapp/azext_containerapp/_up_utils.py#L802) function and this change makes sense now.

Previously, if no local application source was provided, we would still meet this conditional and call the `_get_dockerfile_content` function, and once inside that function, we would've hit the repository conditional since we could have _only_ called this function if the user provided the corresponding source or repository argument.

Because our GitHub Action is going to check for the Dockerfile already, we don't need additional logic in the code to make decisions that are dependent on the repository having a Dockerfile already. I think it makes sense to explicitly handle the local source with Dockerfile case using the result of `_has_dockerfile` and deferring any repository Dockerfile case to the GitHub Action itself.

I would personally leave a comment above the modified lines to signal that we only care about the local source Dockerfile cases and can safely ignore the Dockerfile cases in the repository cases due to its inherent check of the Dockerfile (however you'd like to word that).

Reply:
Added comment to justify why we don't need to look for Dockerfile explicitly in case of `--repo`